src/adapter.ts

```diff
@@ -696,7 +697,16 @@ export class USBAdapter extends EventEmitter implements Adapter {
             const type = this.controlTransferParamsToType(setup, LIBUSB_ENDPOINT_IN);
 
             device.controlTransfer(type, setup.request, setup.value, setup.index, length, (error, buffer) => {
-                if (error) return reject(error);
+                if (error) {
```

Comment:
IMO, `error && error.errno === LIBUSB_TRANSFER_STALL` is more succinct.

Reply:
There are still other possible errors, I would imagine. So the current structure avoids checking for `error` twice.

Reply:
Hmm, until multiple cases appear, I'd prefer it was flattened.

Reply:
Currently if *any* error occurs it will reject the promise. Flattening the if statement to `error && error.errno === LIBUSB_TRANSFER_STALL` means it will not reject if an error other than stall occurs, and instead the promise will resolve with `status` === `ok`, which means it's swallowing the error.

To clarify, I believe this to be how you're suggesting it be written:

```typescript
endpoint.transfer(buffer, error => {
    if (error && error.errno === LIBUSB_TRANSFER_STALL) {
        return resolve({
            bytesWritten: 0,
            status: "stall"
        });
    }

    // Any other error is now swallowed by the resolve with status === ok

    resolve({
        bytesWritten: buffer.byteLength, // hack, should be bytes actually written
        status: "ok" // hack
    });
});
```

I don't think that's a good idea since it opens the possibility of errors being silently hidden. Unless you're suggesting:

```typescript
endpoint.transfer(buffer, error => {
    if (error && error.errno === LIBUSB_TRANSFER_STALL) {
        return resolve({
            bytesWritten: 0,
            status: "stall"
        });
    }

    if (error) return reject(error);

    resolve({
        bytesWritten: buffer.byteLength, // hack, should be bytes actually written
        status: "ok" // hack
    });
});
```

In which case functionally it's the same, I just don't see the point in flattening the if statement and doing a duplicate check on the presence of `error`.

Reply:
Yes, you're right. Sorry :)